inc/datahandlers/user.php

```diff
@@ -888,6 +888,20 @@ function verify_checkfields()
 	}
 
 	/**
+	 * Verifies if the selected timezone is valid.
+	 *
+	 * @return boolean True when valid, false when invalid.
+	 */
+	function verify_timezone()
+	{
+		$user = &$this->data;
+
+		$timezones = get_supported_timezones();
+
+		return array_key_exists($user['timezone'], $timezones);
```

Comment:
@Sama34, we can't move it to `verify_options()` because as I said `timezone` is not an option and the function will only run only if there are options.